### ISSUE TITLE
Error with the LOD Chunks

### ISSUE THREAD

**Issue by JerryBerry12** [glaretechnologies/substrata#7]
Hi,
the new LOD Chunks feature prevents the bike ramp from loading correctly. It cannot actually load until you come close enough to the actual hoops, which means you can't ride a bike up it. My guess to fix it would probably be to implement something in the Tools -> Options windows, with something like "LOD Chunk Load Distance" or something like that. Kind of like object load distance.
Also, I'm not sure if it's the LOD Chunks that are causing this problem, so correct me if I'm wrong.
Double also, love the work you've done here; it looks really good!

**Comment by JerryBerry12** [comment]
Hi, any update? Its been a while

**Comment by Ono-Sendai** [comment]
Yeah this is an issue.  I think it can be fixed by excluding very large objects like the bike ramp from being baked into LOD chunks.

**Comment by JerryBerry12** [comment]
that could probably work. what cpp file deals with making LOD chunks?

**Comment by Ono-Sendai** [comment]
ChunkGenThread

**Comment by JerryBerry12** [comment]
just opened https://github.com/glaretechnologies/substrata/pull/8. I think we should just do this, then for every object we don't want to bake (e.g. it's too large or something) can just be manually specified.

**Comment by Ono-Sendai** [comment]
Ok I have solved this (hopefully) in an automatic way: https://github.com/glaretechnologies/substrata/commit/b09690da33f2ee34baa15830ebb970aeea8c00ce

**Comment by JerryBerry12** [comment]
please consider fixing this by adding the code I have requested in #8. I'm having trouble when running my own server. I am making an LED light show room, and I have too many beams that are flashing at two often intervals. the server keeps trying to bake the chunks every 0.5 seconds, and thus runs out of RAM. In order to get by this, I'm currently building my own copy of the server to put this code into it, so that the issue is fixed. But, I don't want to have to rebuild every time you release a new version.

**Comment by Ono-Sendai** [comment]
How are the LED ligthts flashing?  is it a script of some sort?

**Comment by JerryBerry12** [comment]
yes its a luau script

**Comment by JerryBerry12** [comment]
rn I'm building my own copy of the server but painfully bc the server I'm trying to run it on doesn't have a lot of memory (oracle cloud free tier)

**Comment by Ono-Sendai** [comment]
can u paste the script?    scripted objects shouldn't be baked.  But if a script modfies another object that might be an issue.

**Comment by JerryBerry12** [comment]
Here's a link for a screen recording of the beams before my server crashed.
[https://[jssources.com/substrata_led_wall.mp4](https://jssources.com/substrata_led_wall.mp4)](https://[jssources.com/substrata_led_wall.mp4](https://jssources.com/substrata_led_wall.mp4))

**Comment by JerryBerry12** [comment]
unfortunately cannot post the script right now bc my server is down for building but will do once it's done.
it's building LLVM right now I think thats the longest one

**Comment by JerryBerry12** [comment]
```
--lua
local colors = {
    {1, 0, 0}, -- Red
    {0, 1, 0}, -- Green
    {0, 0, 1}, -- Blue
    {1, 1, 0}, -- Yellow
	{0.7803921568627451, 0.1411764705882353, 0.6941176470588235}, --neon pruple
	
}


function onTimerEvent(object)
    local color = colors[math.random(#colors)]
	mat = object:getMaterial(0)
	mat.colour = Vec3f(color[1], color[2], color[3])
end

createTimer(onTimerEvent, 0.001, true)

```

**Comment by JerryBerry12** [comment]
So something weird happened. Idk if it was a combination of restarting the server and turning off and back on server side scripting, but now it just works. It doesn't bake the led lights. Hmm interesting loophole there. I wonder what caused it

**Comment by JerryBerry12** [comment]
I just spent the whole weekend building the server for nothing 🤦‍♂️

**Comment by JerryBerry12** [comment]
can you pls make an option to turn off LOD chunking entirely? I would really appreciate it, as this feature is pretty much causing nothing but trouble for my server. it just doesn't have enough memory to be constantly updating them every time an object is updated.

**Comment by Ono-Sendai** [comment]
You can add an option if you want!
could add to ServerConfig.
